packages/next/src/build/app-segments/app-segment-config.ts

```diff
@@ -0,0 +1,125 @@
+import { z } from 'next/dist/compiled/zod'
+
+/**
+ * The schema for configuration for a page.
+ *
+ * @internal
+ */
+export const AppSegmentConfigSchema = z.object({
+  /**
+   * The number of seconds to revalidate the page or false to disable revalidation.
+   */
+  revalidate: z
+    .union([z.number().int().nonnegative(), z.literal(false)])
+    .optional(),
+
+  /**
+   * Whether the page supports dynamic parameters.
+   */
+  dynamicParams: z.boolean().optional(),
+
+  /**
+   * The dynamic behavior of the page.
+   */
+  dynamic: z
+    .enum(['auto', 'error', 'force-static', 'force-dynamic'])
+    .optional(),
+
+  /**
+   * The caching behavior of the page.
+   */
+  fetchCache: z
+    .enum([
+      'auto',
+      'default-cache',
+      'only-cache',
+      'force-cache',
+      'force-no-store',
+      'default-no-store',
+      'only-no-store',
+    ])
+    .optional(),
+
+  /**
+   * The preferred region for the page.
+   */
+  preferredRegion: z.union([z.string(), z.array(z.string())]).optional(),
+
+  /**
+   * Whether the page supports partial prerendering. When true, the page will be
+   * served using partial prerendering. This setting will only take affect if
+   * it's enabled via the `experimental.ppr = "incremental"` option.
+   */
+  experimental_ppr: z.boolean().optional(),
+
+  /**
+   * The runtime to use for the page.
+   */
+  runtime: z.enum(['edge', 'nodejs', 'experimental-edge']).optional(),
```

Comment:
```suggestion
  runtime: z.enum(['edge', 'nodejs']).optional(),
```
`experimental-edge` should only be legacy for `pages/` API routes